client/landing/stepper/declarative-flow/internals/steps-repository/link-in-bio-post-setup/index.tsx

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+import { StepContainer, base64ImageToBlob, uploadAndSetSiteLogo } from '@automattic/onboarding';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { useI18n } from '@wordpress/react-i18n';
+import { FormEvent, useEffect, useState } from 'react';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { SITE_STORE } from 'calypso/landing/stepper/stores';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useSite } from '../../../../hooks/use-site';
+import SetupForm from '../link-in-bio-setup/setup-form';
+import type { Step } from '../../types';
+
+import '../link-in-bio-setup/styles.scss';
+
+const LinkInBioPostSetup: Step = function LinkInBioPostSetup( { navigation } ) {
+	const { submit } = navigation;
+	const { __ } = useI18n();
+	const site = useSite();
+
+	const [ siteTitle, setComponentSiteTitle ] = useState( '' );
+	const [ tagline, setTagline ] = useState( '' );
+	const [ invalidSiteTitle, setInvalidSiteTitle ] = useState( false );
+	const [ selectedFile, setSelectedFile ] = useState< File | undefined >();
+	const [ base64Image, setBase64Image ] = useState< string | null >();
+	const [ isLoading, setIsLoading ] = useState< boolean >( false );
+	const [ isSubmitError, setIsSubmitError ] = useState< boolean >( false );
+
+	const { saveSiteSettings } = useDispatch( SITE_STORE );
+
+	useEffect( () => {
+		setComponentSiteTitle( site?.name || '' );
+		setTagline( site?.description || '' );
+	}, [ site ] );
+
+	useEffect( () => {
+		setIsSubmitError( false );
+	}, [ siteTitle, tagline, selectedFile, base64Image ] );
+
+	const handleSubmit = async ( event: FormEvent ) => {
+		event.preventDefault();
+
+		if ( ! siteTitle.trim().length ) {
+			setInvalidSiteTitle( true );
+			return;
+		}
+
+		setIsLoading( true );
```

Comment:
Shouldn't this be inside the `if (site)` check? Otherwise it will be stuck forever if somehow someone managed to click before having a site.

Reply:
@alshakero - That makes sense. For related reasons, I also moved the following `setIsLoading( false );` and `submit?.();` (now on line 61-62) inside the `if ( site )` check. The idea is that if if site doesn't exist, we don't want to take action, including changing loading state or actually submitting and advancing to next screen.